src/catalog/src/durable.rs

```diff
@@ -80,10 +69,6 @@ pub struct BootstrapArgs {
 
 pub type Epoch = NonZeroI64;
 
-pub(crate) fn epoch_checked_increment(epoch: Epoch) -> Option<Epoch> {
-    epoch.get().checked_add(1).and_then(Epoch::new)
-}
-
 /// An API for opening a durable catalog state.
 ///
 /// If a catalog is not opened, then resources should be release via [`Self::expire`].
```

Comment:
I'm still on the fence about keeping/removing the trait. It might be a nice abstraction to prevent higher layers from relying on the durable implementation. It also may help in the future if we ever decide to switch again. On the other hand it might be a bit over-engineered now that there's only a single implementation. I'm not really sure, but I don't think it needs to be removed in this PR.